src/utils.js

```diff
@@ -675,6 +676,22 @@ function removeStringsFromList(list, stringsToRemove) {
   return list.filter((item) => !removalSet.has(item));
 }
 
+/**
+ * Remove strings from a list, matching in a case-insensitive manner
+ * @param {Array<string>} list
+ *  List of strings to remove from
+ * @param {Array<string>} stringsToRemove
+ *  Strings that should be removed from list
+ * @returns {Array<string>}
+ */
+function removeStringsFromListCaseInsensitive(list, stringsToRemove) {
+  const lowercaseMap = fromPairs(list.map(str => [str.toLowerCase(), str]));
+  const lowercaseList = list.map(str => str.toLowerCase());
+  const lowercaseToRemove = stringsToRemove.map(str => str.toLowerCase());
+  const removedLowercase = removeStringsFromList(lowercaseList, lowercaseToRemove);
+  return removedLowercase.map(str => lowercaseMap[str]);
```

Comment:
Good solution. Do you think we could simplify it even more for readability?
Maybe something similar to this?
```
const lowerCaseRemoveList = stringsToRemove.map(str => str.toLowerCase());
const result = list.filter(str => !lowerCaseRemoveList.includes(str.toLowerCase());
return result;
```

Reply:
That is a more concise and readable implementation.  However, its performance isn't as good for large datasets, being approximately O(n^2) as opposed to the current O(n).

I did go ahead and add comments on the current implementation, which should help future readers understand what's going on.

Reply:
Understanding where this is used, that makes sense.